src/textual/message_pump.py

```diff
@@ -542,6 +610,7 @@ async def post_message(self, message: Message) -> bool:
             return False
         if not self.check_message_enabled(message):
             return True
+        message._prevent.update(self._get_prevented_messages())
```

Comment:
This line confuses me a bit.
Is this the line that had to do with the extra thing you mentioned you ended up figuring out you had to do, in that as the message bubbles up, you need to continuously update the prevent stack with these things?

If so, do you want to write something with regard to that? If this only came to you as an afterthought or when some tests broke, it may be a good idea to comment above (or in the docstring) why we do this kind of update.